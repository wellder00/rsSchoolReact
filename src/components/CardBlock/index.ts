export { default as CardBlock } from './CardBlock';
