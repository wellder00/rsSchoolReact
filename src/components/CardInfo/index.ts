export { default as CardInfo } from './CardInfo';
