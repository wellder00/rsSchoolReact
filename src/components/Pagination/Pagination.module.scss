.wrapper {
  padding-top: 20px;
}

.currentPage {
  padding: 0 20px;
  font-size: 40px;
  font-family: 'Metal Mania', sans-serif;
}

.rightArrow {
  transform: rotate(180deg);
}

.arrow {
  filter: drop-shadow(0 0 20px red);
  cursor: pointer;
  border-radius: 100%;
  background-color: rgb(248, 211, 38);
  width: 60px;
  height: 60px;
  &:hover {
    background-color: rgb(231, 191, 10);
  }

  &:active {
    background-color: rgb(197, 164, 20);
  }
}

.disabled {
  border-radius: 100%;
  background-color: grey;
  width: 60px;
  height: 60px;
}
