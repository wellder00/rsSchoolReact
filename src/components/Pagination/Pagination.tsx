import styles from './Pagination.module.scss';
import arrow from '../../assets/images/arrow.png';
import { useAppSelector } from '../../Hooks/reduxHooks';

type Props = {
  onChangePage: (value: boolean) => void;
};

const Pagination: React.FC<Props> = ({ onChangePage }) => {
  const currentPage = useAppSelector((state) => state.itemsAmount.currentPage);
  const lastPage = useAppSelector((state) => state.itemsAmount.lastPage);

  const isPrevButtonDisabled = +currentPage <= 1;

  const isNextButtonDisabled = +currentPage === +lastPage;
  const prevButtonClass = isPrevButtonDisabled ? styles.disabled : styles.arrow;
  const nextButtonClass = isNextButtonDisabled ? styles.disabled : styles.arrow;

  return (
    <div className={styles.wrapper}>
      <button
        disabled={isPrevButtonDisabled}
        onClick={() => onChangePage(false)}
        className={prevButtonClass}
      >
        <img className={styles.leftArrow} src={arrow} alt="leftArrow" />
      </button>

      <span className={styles.currentPage}>{currentPage}</span>

      <button
        disabled={isNextButtonDisabled}
        onClick={() => onChangePage(true)}
        className={nextButtonClass}
      >
        <img className={styles.rightArrow} src={arrow} alt="rightArrow" />
      </button>
    </div>
  );
};

export default Pagination;
