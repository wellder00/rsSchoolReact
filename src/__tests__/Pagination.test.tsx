import { render, screen } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import { Pagination } from '../components/Pagination';

describe('Pagination', () => {
  it('Check rightArrow', () => {
    const pages = {
      offset: 0,
      currentPage: 1,
      lastPage: 8,
    };
    const { asFragment } = render(
      <MemoryRouter>
        <Pagination onChangePage={() => false} pages={pages} />
      </MemoryRouter>
    );

    expect(asFragment()).toMatchSnapshot();
    expect(screen.getByAltText(/rightArrow/i)).toBeInTheDocument();
  });
  it('Check leftArrow', () => {
    const pages = {
      offset: 0,
      currentPage: 1,
      lastPage: 8,
    };
    render(
      <MemoryRouter>
        <Pagination onChangePage={() => false} pages={pages} />
      </MemoryRouter>
    );

    expect(screen.getByAltText(/leftArrow/i)).toBeInTheDocument();
  });
});
