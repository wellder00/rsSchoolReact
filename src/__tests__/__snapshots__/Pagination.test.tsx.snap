// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Pagination > Check rightArrow 1`] = `
<DocumentFragment>
  <div
    class="_wrapper_08165b"
  >
    <button
      class="_disabled_08165b"
      disabled=""
    >
      <img
        alt="leftArrow"
        class="_leftArrow_08165b"
        src="/src/assets/images/arrow.png"
      />
    </button>
    <span
      class="_currentPage_08165b"
    >
      1
    </span>
    <button
      class="_arrow_08165b"
    >
      <img
        alt="rightArrow"
        class="_rightArrow_08165b"
        src="/src/assets/images/arrow.png"
      />
    </button>
  </div>
</DocumentFragment>
`;
